docs/path-resolution.rst

```diff
@@ -0,0 +1,465 @@
+
+       solc contract.sol /usr/local/dapp-bin/token.sol
+
+   The source unit name of a file loaded this way is simply the specified CLI path with
+   platform-specific separators converted to forward slashes.
+
+   .. index:: standard JSON
+
+#. **Standard JSON**
+
+   When using the :ref:`Standard JSON <compiler-api>` API (via either the `JavaScript interface
+   <https://github.com/ethereum/solc-js>`_ or the ``--standard-json`` command-line option)
+   you provide input in JSON format, containing, among other things, the content of all your source
+   files:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "contract.sol": {
+                   "content": "import \"./util.sol\";\ncontract C {}"
+               },
+               "util.sol": {
+                   "content": "library Util {}"
+               },
+               "/usr/local/dapp-bin/token.sol": {
+                   "content": "contract Token {}"
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The ``sources`` dictionary becomes the initial content of the virtual filesystem and its keys
+   are used as source unit names.
+
+#. **Standard JSON (via import callback)**
+
+   With Standard JSON it is also possible to tell the compiler to use the import callback to obtain
+   the source code:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "/usr/local/dapp-bin/token.sol": {
+                   "urls": [
+                       "/projects/mytoken.sol",
+                       "https://example.com/projects/mytoken.sol"
+                   ]
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   If an import callback is available, the compiler will give it the source unit names specified in
+   ``urls`` one by one, until one is loaded successfully or the end of the list is reached.
+
+   The source unit names are determined the same way as when using ``content`` - they are keys of
+   the ``sources`` dictionary and the content of ``urls`` does not affect them in any way.
+
+   .. index:: standard input, stdin, <stdin>
+
+#. **Standard input**
+
+   On the command line it is also possible to provide the source is by sending it to compiler's
+   standard input:
+
+   .. code-block:: bash
+
+       echo 'import "./util.sol"; contract C {}' | solc -
+
+   ``-`` used as one of the arguments instructs the compiler to place the content of the standard
+   input in the virtual filesystem under a special source unit name: ``<stdin>``.
+
+Once the VFS is initialized, additional files can still be added to it only through the import
+callback.
+
+.. index:: ! import; path
+
+Imports
+=======
+
+The import statement specifies an *import path*, which, after being lightly processed, becomes a
```

Comment:
Maybe we could also add here `import "<import-path>"`, to ensure that everyone understands the definition of an import path.

Reply:
Done. I added an example with two imports and described which part is the import path and how it relates to the source unit name.